### PR TITLE
quic: handle RESET_STREAM frame

### PR DESCRIPTION
**Pull request by ripatel-fd**
See RFC 9000 Section 19.4

### CODE REVIEW

File: src/waltz/quic/fd_quic.c [hunk 1]
```diff
@@ -6253,17 +6253,82 @@ fd_quic_frame_handle_ecn_counts_frag(
...
+  /* Notify client that local recv interest is reset */
+  stream->state |= FD_QUIC_STREAM_STATE_RX_FIN;
+  if( stream->state & FD_QUIC_STREAM_STATE_TX_FIN || !bidir ) {
+    fd_quic_stream_free( quic, conn, stream, FD_QUIC_NOTIFY_RESET );  /* calls notify */
+  } else {
+    fd_quic_cb_stream_notify( quic, stream, stream->context, FD_QUIC_NOTIFY_RESET );
+  }
+
+ack:
+  pkt->ack_flag |= ACK_FLAG_RQD;
```
Comment (ripatel-fd) [inline src/waltz/quic/fd_quic.c:6330]: I'm only setting the ACK flag on successful handling of the frame. 
Should I also set it when abandoning the connection with an error?

Comment (nbridge-jump) [reply src/waltz/quic/fd_quic.c:6330]: I doubt it makes much difference. 
As a technicality and in general, it should probably ACK only-if all the frames were handled successfully. If any frames were faulty or ignored/unprocessed, then none of the frames should have any effect, and the ACK should not be sent. In any case, if some frame were unable to be processed, we should probably close the connection with an error, like internal-error or protocol-violation, or some more specific error.

File: src/waltz/quic/fd_quic.c [hunk 2]
```diff
@@ -6253,17 +6253,82 @@ fd_quic_frame_handle_ecn_counts_frag(
...
+    if( FD_UNLIKELY( !bidir ) ) {
+      /* This stream type can only be initiated locally and is uni-
+         directional local-to-peer.  See RFC 9000 Section 19.4:
+         > An endpoint that receives a RESET_STREAM frame for a
+         > send-only stream MUST terminate the connection with error
+         > STREAM_STATE_ERROR. */
+      fd_quic_conn_error( conn, FD_QUIC_CONN_REASON_STREAM_STATE_ERROR, __LINE__ );
+      return FD_QUIC_PARSE_FAIL;
+    } else {
+      /* No-op, can't reason about peer-initiated bidirectional stream
```
Comment (nbridge-jump) [inline src/waltz/quic/fd_quic.c:6289]: This is in the local_initiated+bidir branch

File: src/waltz/quic/fd_quic.c (hunk 1 above)
Comment (nbridge-jump) [reply src/waltz/quic/fd_quic.c:6330]: In this case, I would not send ACK

**Comment by ripatel-fd** [comment]
Before merging, I have to add a unit test for this one

**Review by nbridge-jump — APPROVED** [review]

**Comment by ripatel-fd** [comment]
Closing this because I'm about to send a PR that makes this redundant.

**Pull request closed without merging**